From 49b854fe9f90320aa0afa97aeea3789a663b880c Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Sat, 11 Oct 2025 13:22:34 +0200
Subject: [PATCH 089/116] pwm: mediatek: fix regoffset changed to
 channel-base/width

c95ab56a7ad6 pwm: mediatek: Simplify representation of channel offsets
---
 drivers/pwm/pwm-mediatek.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pwm/pwm-mediatek.c b/drivers/pwm/pwm-mediatek.c
index e56ea699c110..64ae12f4f544 100644
--- a/drivers/pwm/pwm-mediatek.c
+++ b/drivers/pwm/pwm-mediatek.c
@@ -489,7 +489,8 @@ static const struct pwm_mediatek_of_data mt7986_pwm_data = {
 static const struct pwm_mediatek_of_data mt7987_pwm_data = {
 	.num_pwms = 3,
 	.pwm45_fixup = false,
-	.reg_offset = mtk_pwm_reg_offset_v3,
+	.chanreg_base = 0x100,
+	.chanreg_width = 0x100,
 };
 
 static const struct pwm_mediatek_of_data mt7988_pwm_data = {
-- 
2.39.5

